lib/teleterm/services/desktop/desktop.go

```diff
@@ -217,16 +217,20 @@ func isClientMessageAllowed(msg tdp.Message) error {
 	}
 }
 
-// serverInterceptor intercepts and processes messages sent from the server to the client.
-type serverInterceptor struct {
+// fsRequestHandler handles file system messages sent from the server to the client.
+//
+// If a message is a file system request, it is handled internally via DirectoryAccess instead of being
+// forwarded to the Electron app. The response is then sent back to the server.
+// If the message is not a file system request, it is returned as-is to be forwarded to the client.
```

Comment:
Yeah, that's clearer.

It'd be nice to add integration tests for this at some point, but I'm not sure how feasible that is.

Reply:
It would be nice, but yeah, setting this up would be quite complex